lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1046,6 +1151,29 @@ SubExprInfo ExprEmitter::visitComb(MuxOp op) {
   return {Conditional, signedness};
 }
 
+SubExprInfo ExprEmitter::visitComb(StructExtractOp op) {
+  auto structPrec = emitSubExpr(op.input(), Selection);
+  os << '.' << op.field();
+  return {Selection, structPrec.signedness};
```

Comment:
Either this or unsigned unconditionally.  the returned sign is the sign of the op and the field itself has no sign.

Reply:
The RTL dialect has an open type system, so the fields do have signedness: #469 #397. These are not implemented, so I'm ok with unconditionally unsigned for the time being. We can clean this up when we get around to fixing these two issues.

Reply:
However, this is ExportVerilog which has a closed world assumption.

Reply:
Yeah, ExportVerilog has a closed world assumption.

That said, are you sure verilog propagates signedness in the logical way?  I would guess it does something dumb like making the result always unsigned.  it would be good to check this, the same as array indexing.

Reply:
The standard integer type lowers to unsigned verilog types.

Reply:
We are explicitly generating unsigned structures.  Verilog support signed structures, but we are not generating such craziness.

Reply:
I'm saying the integer field inside a struct may be signed, unsigned, or signless. Not the struct itself, which I agree we shouldn't support signed structs as they are -- as you accurately state -- crazy. Given that the RTL dialect is an open type system, we need to respect the signedness of the fields. See the discussion in #469 .

@lattner As usual, I can't find anything in the spec which explicitly discusses the signedness of struct fields. There is something in 11.8.1 "The sign and size of any self-determined operand are determined by the operand itself and
independent of the remainder of the expression." So far as I can deduce / guess member accesses are self-determined expressions, though if you look at the definition of "self-determined" expressions (buried in 11.6.1) you'll note that "self-determined" is context-dependent. *Sigh*. Again, I think it's a good assumption that verilog respects the signedness of struct fields given that it allows them to have the signed/unsigned attributes. But we'd need some experiments to be sure-ish.